presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralEncoder.java

```diff
@@ -99,6 +101,14 @@ public Expression toExpression(Object object, Type type)
             return new Cast(new NullLiteral(), type.getTypeSignature().toString(), false, true);
         }
 
+        if (type.equals(TINYINT)) {
+            return new GenericLiteral("TINYINT", object.toString());
+        }
+
+        if (type.equals(SMALLINT)) {
+            return new GenericLiteral("SMALLINT", object.toString());
```

Comment:
any tests?

Reply:
Perhaps this commit should be squashed with "Add test cases for GenericLiterals"